test/unit/Optimize.ml

```diff
@@ -1295,6 +1295,340 @@ let%expect_test "unroll nested loop" =
 
       } |}]
 
+let%expect_test "unroll nested loop 2" =
+  let _ = gensym_reset_danger_use_cautiously () in
+  let ast =
+    Parse.parse_string Parser.Incremental.program
+      {|      model {
+                for (i in 1:2)
+                  for (j in i:4)
+                    for (k in j:9)
+                       print(i, j, k);
+                   }
+      |}
+  in
+  let ast = semantic_check_program ast in
+  let mir = Ast_to_Mir.trans_prog "" ast in
+  let mir = static_loop_unrolling mir in
+  Fmt.strf "@[<v>%a@]" pp_typed_prog mir |> print_endline ;
+  [%expect
+    {|
+      functions {
+
+      }
+
+      input_vars {
+
+      }
+
+      prepare_data {
+
+      }
+
+      log_prob {
+        {
+          {
+            {
+              {
+                FnPrint__(1, 1, 1);
```

Comment:
Should these really all be so deeply nested in blocks?

Also these tests would be much more readable if they just printed the part under test, e.g. the log_prob here... taking a crack at that by updating the pretty printer to not print empty things and reduce the number of spare lines.